pkg/generate/app/cmd/newapp.go

```diff
@@ -486,6 +490,17 @@ func (c *AppConfig) search(components app.ComponentReferences) error {
 	return errors.NewAggregate(errs)
 }
 
+func (c *AppConfig) detectPartialMatches(components app.ComponentReferences) error {
```

Comment:
it's fixing a bug with how we reported partial matches previously.  We previously only reported partial matches for image builders, we should be reporting them for any image object.

we get here in any scenario where there is an inexact match.  Again this is just a refactoring that applies this check to all image matches instead of only image matches for source-repos.